week1/day_4_plotting_exercises.R

```diff
@@ -0,0 +1,109 @@
+# Session 3.3.1, exercise 1, 2, and 3
+library(tidyverse)
+
+ggplot(data = mpg) + 
+  geom_point(mapping = 
+               aes(x = displ, y = hwy, color = "blue"))
+
+# 1. The points aren't blue because the argument color is a string blue that is treated as an aes. 
+# This should be a mapping between a variable and a value. "blue" is not a value that represents the color blue. 
+# If you place, color = "blue" outside the aes, they the graph would be blue. 
+
+# 2. categorical variables: manufacturer, model, trans, drv, fl, class
+# continuous variables are: displ, year, cyl, cty, hey
+
+# 3. 
+ggplot(data = mpg) + 
+  geom_point(mapping = 
+               aes(x = displ, y = hwy, color = year))
+# a continuous variable for color uses a scale of colors from light to dark.
+ggplot(data = mpg) + 
+  geom_point(mapping = 
+               aes(x = displ, y = hwy, size = year))
+# a continuous variable for size, changes how big or small the points on the graph are. 
+ggplot(data = mpg) + 
+  geom_point(mapping = 
+               aes(x = displ, y = hwy, shape = year))
+# A continuous variable can not be mapped to shape.
+# I think the color and size are not the best to use for continuous variables. 
+# The reader of the graph needs to keep on going back to the legend to understand the graph. 
+
+# Session 3.5.1, exercise 1 and 4
+# 1. 
+ggplot(data = mpg) + 
+  geom_point(mapping = aes(x = displ, y = hwy)) + 
+  facet_wrap(~ displ, nrow = 2)
+# R lets you wrap a continuous variable, however, the results are not clearly readble. 
+# I would not do this. 
+
+# 4. 
+ggplot(data = mpg) + 
+  geom_point(mapping = aes(x = displ, y = hwy)) + 
+  facet_wrap(~ class, nrow = 2)
+# Below class is wrapped to a color:
+ggplot(data = mpg) + 
+  geom_point(mapping = aes(x = displ, y = hwy, color = class)) 
+# The color is good when there are not so many categories 
+# (or else you might run out of colors - or have very similar looking colors)
+# facetting the class increases the number of points and categories readable. 
+# We don't have much overlap of different or the same color points. 

```

Comment:
yeah I agree. You can find color palettes online to solve the similar looking colors problem (or I think there's a package somewhere that can generate decent ones) but it doesn't solve the problem with it just being cluttered.